internal/filters/main.go

```diff
@@ -0,0 +1,101 @@
+/*
+A CLI->Go template shim used to render arbitrary sprig functions.
+
+To Run:
+> go run main.go sprigFunctionName sprigFunctionArguments...
+
+Example:
+> go run main.go trim "      some string      "
+
+This program takes textual input and attempts to resolve the sprigFunctionArguments types through inspecting
+sprigFunctionName's signature.  Bad input results in fatal exit with error code 1.  Debug output is intentionally
+suppressed as stdout is utilized by the calling program.
+*/
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Masterminds/sprig"
+	"os"
+	"reflect"
+	"strconv"
+)
+
+const fatalExitReturnCode = 1
+const funcArgsCLIArgsFirstIndex = 2
+const funcNameCLIArgsIndex = 1
+const funcReturnOutputIndex = 0
+
+// Wrapper to call a Golang function through reflection.
+func CallFunc(funcMap map[string]interface{}, name string, arguments ... interface{}) (result []reflect.Value,
+	err error) {
+
+	function := reflect.ValueOf(funcMap[name])
+	if len(arguments) != function.Type().NumIn() {
+		err = errors.New("number of arguments is not adapted")
+		return
+	}
+	in := make([]reflect.Value, len(arguments))
+	for k, param := range arguments {
+		in[k] = reflect.ValueOf(param)
+	}
+	result = function.Call(in)
+	return
+}
+
+// Exits 1 with an appropriate message.
+func fatalError(message string) {
+	fmt.Fprintln(os.Stderr, "ERROR: " + message)
+	os.Exit(fatalExitReturnCode)
+}
+
+// Ensures that a sprig/template function was provided as a CLI argument.
+func checkForFuncName(args *[]string) {
+	if len(*args) < 2 {
+		fatalError("No Sprig Function name was provided")
+	}
+}
+
+// Invokes the CLI shim.  The first argument is the sprig/template function name.  The remaining arguments are the
+// sprig/template function arguments.
+func main() {
+	args := os.Args
+	// This would indicate a function with no funcName argument.
+	checkForFuncName(&args)
+
+	funcName := args[funcNameCLIArgsIndex]
+	funcArgs := args[funcArgsCLIArgsFirstIndex:]
+
+	// Loads only the text/template and Sprig Function Map.
+	funcMap := sprig.TxtFuncMap()
```

Comment:
I would put this in its own file and not have logic embeded in main.
Define a type for funcs like `type funcs map[string]reflect.Value` and then add methods to it.But if you are planning to throw this away and just for demo purpose, I am okay with that.

Reply:
I agree with you in principal, but I would like to take the route that this is "throw away demo code".  If our stakeholders like what we show them, then we can consider replacing this with a more robust solution, such as https://github.com/tsg/gotpl.  This was very much a first pass, and we can discuss further investment with David, Andrew and others.